app/controllers/create-service/select-organisation-type/select-organisation-type.controller.js

```diff
@@ -0,0 +1,62 @@
+const _ = require('lodash')
+const paths = require('../../../paths')
+const { response } = require('../../../utils/response')
+const {
+  validateMandatoryField,
+  SERVICE_NAME_MAX_LENGTH,
+  validateOptionalField
+} = require('../../../utils/validation/server-side-form-validations')
+const logger = require('../../../utils/logger')(__filename)
+
+function get (req, res) {
+  const createServiceState = _.get(req, 'session.pageData.createService', {})
+  if (!createServiceState.current_name) {
+    logger.info('Page accessed out of sequence, redirecting to my-services/create')
+    return res.redirect(paths.serviceSwitcher.create.index)
+  }
+  const context = {
+    ...createServiceState,
+    back_link: paths.serviceSwitcher.create.index,
+    submit_link: paths.serviceSwitcher.create.index
+  }
+  _.unset(req, 'session.pageData.createService.errors')
+  return response(req, res, 'services/select-org-type', context)
+}
+
+function post (req, res) {
+  _.set(req, 'session.pageData.createService', {
+    current_name: req.body['service-name'],
```

Comment:
might be simpler if both of these are `'service-name'`/`'service-name-cy'` vs mixing in `current_name`/`current_name_cy`